dozer-types/protos/cloud.proto

```diff
@@ -52,93 +40,103 @@ service DozerCloud {
+  repeated File files = 2;
+  repeated Secret secrets = 3;
+  bool allow_incompatible = 4;
+  InfraRequest infra = 5;
 }
-message DeploymentStatusResponse {
-  repeated DeploymentHealth k8_deployments = 1;
-  repeated string errors = 2;
-  map<uint32, uint32> versions = 3;
-  uint32 current_version = 4;
+
+message InfraRequest {
+  int32 num_api_instances = 2;
 }
 
-message DeploymentHealth {
-  string name = 1;
-  string status = 2;
-  string created_at = 3;
-  int32 desired = 4;
-  int32 available = 5;
+message DeployStep {
+  bool is_completed = 1;
+  string text = 2;
+  uint32 current_step = 3;
+  optional string error = 4;
 }
 
-message AppConnectionRequest {
-  string app_id = 1;
-  oneof input {
-    Connection connection = 2;
-    string connection_yaml = 3;
+message DeployAppResponse {
+  oneof result {
+    AppResponse app_created = 1;
+    DeployStep step = 2;
+    string log_message = 4;
   }
+  bool completed = 5;
+  optional string error = 6;
 }
 
-message CreateSourceRequest {
+message AppResponse {
   string app_id = 1;
-  Source source = 2;
-  Connection connection = 3;
+  AppConfig app = 2;
+  optional string repository_url = 3;
+  repeated DeploymentInfo deployments = 4;
+  google.protobuf.Timestamp created_at = 5;
+  google.protobuf.Timestamp updated_at = 6;
 }
 
-message ListEndpointRequest {
-  string app_id = 1;
-  optional uint32 limit = 2;
-  optional uint32 offset = 3;
+message ListAppRequest {
+  optional uint32 limit = 1;
+  optional uint32 offset = 2;
+  optional string name = 3;
+  optional string uuid = 4;
+  optional string order_by = 5;
+  optional bool desc = 6;
 }
-
-message ListEndpointResponse {
-    repeated EndpointResponse endpoints = 1;
-    Pagination pagination = 2;
+message ListAppResponse {
+  repeated AppResponse apps = 1;
+  Pagination pagination = 2;
 }
 
-message EndpointResponse {
-  string id = 1;
-  Endpoint endpoint = 2;
-  string yaml_content = 3; 
+message UpdateAppRequest {
+  string app_id = 1;
+  repeated File files = 2;
 }
 
-message ListSourceRequest {
+message DeleteAppRequest {
   string app_id = 1;
-  optional uint32 limit = 2;
-  optional uint32 offset = 3;
 }
+message DeleteAppResponse { bool success = 1; }
 
-message ListSourceResponse {
-    repeated SourceResponse sources = 1;
-    Pagination pagination = 2;
-}
+message GetAppRequest { optional string app_id = 1; }
 
-message SourceResponse {
-  string id = 1;
-  Source source = 2;
-  string yaml_content = 3;
+message AppHealthRequest {
+  string app_id = 1;
 }
-
-message ListAppConnectionRequest {
+message AppHealthResponse {
+  bool healthy = 1;
+}
+message GetResourcesRequest {
   string app_id = 1;
-  optional uint32 limit = 2;
-  optional uint32 offset = 3;
+  // if not specified, will get latest version
+  optional uint32 version = 2;
+}
+message ResourcesResponse {
+  repeated DeploymentResource resources = 1;
+  map<uint32, uint32> versions = 3;
+  uint32 current_version = 4;
 }
 
+message DeploymentResource {
```

Comment:
Looks like renamed from `DeploymentHealth`. Why remove `DeploymentHealth::status`?

Reply:
@Jesse-Bakker's recent PR introduces Job type for Applications. This complicates things a little bit as payload for `Job` and `Deployment` are a little different through APIs. So we can delegate status calculation to the frontend.

Reply:
I remember `status` was from k8s resource status. It's not returned to frontend?

Reply:
not in the same form. It was being transformed.

Reply:
Yes, I meant that frontend now doesn't have the information that was used to calculate `status`. It only has `desired`, `available`, etc.

Reply:
https://github.com/getdozer/dozer-frontend/pull/71/files#diff-bd9ee897c857753c2ace2bdb9e4d7aefcaed16bd93c86e9bd6e8e031a76d1e98
This wasnt comprehensive before.

Reply:
I'm not sure about what you mean.

As far as I can see, new `status` is calculated based on `desired` and `available`: https://github.com/getdozer/dozer-frontend/pull/71/files#diff-bd9ee897c857753c2ace2bdb9e4d7aefcaed16bd93c86e9bd6e8e031a76d1e98R55
While old `status` is calculated based on pod conditions: https://github.com/getdozer/dozer-cloud/pull/570/files#diff-2579a5f330b2b0551e29104778b207398caa2493197ee3d5c5ca6b0067917daaL11

Reply:
Which is exactly what i was trying to avoid to be done on the backend.